src/v/cluster_link/service.cc

```diff
@@ -74,6 +74,41 @@ struct shard_report_reducer {
     }
     std::optional<result_t> result;
 };
+
+struct shard_link_report_reducer {
+    using result_t = ::cluster_link::rpc::shadow_link_status_report_response;
+    void operator()(result_t shard_result) {
+        if (!result) {
+            result = std::move(shard_result);
+            return;
+        }
+        if (result->err_code != ::cluster_link::errc::success) {
+            // once we have an error, just keep it
+            return;
+        }
```

Comment:
I wonder if we should do this.. (if the shard_result is successful, we can still update the reports with the results from the shard, even tho the whole report is partial).